src/core/os_macos.c

```diff
@@ -53,11 +57,38 @@ void os_sleep(double seconds) {
 }
 
 void os_request_permission(os_permission permission) {
-  //
+  if (permission == OS_PERMISSION_AUDIO_CAPTURE) {
+    // If on old OS without permissions check, permission is implicitly granted
+    if(![AVCaptureDevice respondsToSelector:@selector(authorizationStatusForMediaType:)]) {
+      if(state.onPermissionEvent) state.onPermissionEvent(permission, true);
+    }
+    
+    switch ([AVCaptureDevice authorizationStatusForMediaType:AVMediaTypeAudio]) {
+      case AVAuthorizationStatusAuthorized:
+        if(state.onPermissionEvent) state.onPermissionEvent(permission, true);
+        return;
+      case AVAuthorizationStatusNotDetermined:
+        [AVCaptureDevice
+          requestAccessForMediaType:AVMediaTypeAudio
+          completionHandler:^(BOOL granted) {
+            dispatch_async(dispatch_get_main_queue(), ^{
+              if(state.onPermissionEvent) state.onPermissionEvent(permission, granted);
+            });
+          }
+        ];
+        return;
+      case AVAuthorizationStatusDenied:
+        if(state.onPermissionEvent) state.onPermissionEvent(permission, false);
+        return;
+      case AVAuthorizationStatusRestricted:
+        if(state.onPermissionEvent) state.onPermissionEvent(permission, false);
+        return;
+    }
+  }
```

Comment:
Nits here:

- Would rather use `break` instead of `return` when nothing is below.
- `if (` should have a space before the opening paren.